File: Assignment_14/question4.c
////////////////////////////////////////////////////////////////////////////////
//
//  File Name   : question4.c
//  Description : Alternate positive and negative number rows.
//
//  Problem     : Accept number of rows and number of columns from user and
//                display below pattern.
//
//                Input: iRow = 5, iCol = 5
//                Output:
//                 1  2  3  4  5
//                -1 -2 -3 -4 -5
//                 1  2  3  4  5
//                -1 -2 -3 -4 -5
//                 1  2  3  4  5
//
//  Author      : Asawari Jadhav
//  Date        : 20/05/2025
//
////////////////////////////////////////////////////////////////////////////////

#include <stdio.h>

void Pattern(int iRow, int iCol)
{
  int i = 0, j = 0;
  int num = 0;

  for (i = 1; i <= iRow; i++)
  {
    for (j = 1; j <= iCol; j++)
    {
      num = j;
      if (i % 2 != 0)
        printf("%d\t", num);
      else
        printf("-%d\t", num);
    }
    printf("\n");
  }
}

int main()
{
  int iValue1 = 0, iValue2 = 0;

  printf("Enter number of rows and columns: ");
  scanf("%d %d", &iValue1, &iValue2);

  Pattern(iValue1, iValue2);
  return 0;
}

File: Assignment_49/question4.java
//////////////////////////////////////////////////////////////////////////////////////////////
//
//  File Name     : question4.java
//  Description   : Accept string from user and display pattern
//  Problem       : Combine decreasing and increasing pattern using *
//
//  Author        : Asawari Jadhav
//
//  Date          : 21-08-2025    
// 
//////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Pattern
{
    public void DisplayPattern(String str)
    {
        int len = str.length();
        
        for (int i = 0; i < len; i++)
        {
            for (int j = 0; j < len; j++)
            {
                if (j < len - i)
                {
                    System.out.print(str.charAt(j) + " ");
                }
                else
                {
                    System.out.print("* ");
                }
            }
            System.out.println();
        }

        for (int i = len - 2; i >= 0; i--)
        {
            for (int j = 0; j < len; j++)
            {
                if (j < len - i)
                {
                    System.out.print(str.charAt(j) + " ");
                }
                else
                {
                    System.out.print("* ");
                }
            }
            System.out.println();
        }
    }
}

public class question4
{
    public static void main(String args[])
    {
        Scanner sobj = new Scanner(System.in);
        System.out.print("Enter a string : ");
        String str = sobj.nextLine();

        Pattern pobj = new Pattern();
        pobj.DisplayPattern(str);

        
    }
}
